packages/vm/lib/evm/eei.ts

```diff
@@ -564,7 +564,12 @@ export default class EEI {
       this._lastReturned = results.execResult.returnValue
     }
 
-    if (!results.execResult.exceptionError) {
+    console.log(results.execResult.exceptionError)
+
+    if (
+      !results.execResult.exceptionError ||
+      results.execResult.exceptionError.error === ERROR.CODESTORE_OUT_OF_GAS
```

Comment:
This ensures that, IF we have a CODESTORE_OUT_OF_GAS (Frontier only!), we treat it as "if the contract was created" (we know that it was not created). This means that we actually put the "created address" on stack as return value on the call frame we are returning to. (This ensures that this 0x83.. (empty contract) actually gets the 0.03 ether, instead of it being sent to 0x00.., (the value which is put on stack in case we are on Homestead if contract fails to be created)).